main.cpp

```diff
@@ -89,17 +90,48 @@ struct Cat
         std::string disposition;
         int numLegs = 4;
         int numWhiskers {24};
-        std::string catBreed = "Tabby";
-        
+        std::string catBreed = "Tabby";       
+
         void feed(bool isHungry);
         void pet(bool isAffectionate, float minutesToPet);
         void roamNeighborhood(bool rainyWeather, float avgCatTravels = 20.f);
+        void roll(int numRolls, int happyIdxOutOfTen = 0)
+        {
+            auto hi = happyIdxOutOfTen;
+            while (numRolls < 100)
+            {  
+                for(hi = 1; hi <= 9; ++hi)
+                {
+                    numRolls+=10;
+                    std::cout << "Kitten rolls " << numRolls  << " times on the floor."  << std::endl; 
+                    
+                }
+            break;
+            }
+            std::cout << "Kitten is feeling too dizzy to roll anymore." << std::endl;   
+            
+        }
     };
 
     void scratchFurniture(bool isChair = false, int numberOfFurniture = 10);
     void makeNoise(bool foodBowlEmpty = true);
     bool sleep(bool isAsleep = true);
     void typicalCat(std::string, int, float);
+    void gainWeight(int weight) 
+    {
+        int overweight = 20;
+        auto w = weight;
+        while (weight < overweight)
+        {
+            for(w = 1; w <= 3; ++w)
+            {
+            dailyFoodIntake += 1.f;
+            std::cout << "Kitty eating more and more every day" << std::endl;
+            }
+        break;
+        }
```

Comment:
Fix this indentation.
remove the `break`, because it prevents the `while` loop from looping.
- [ ] revise